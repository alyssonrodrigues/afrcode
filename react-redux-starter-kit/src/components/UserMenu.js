import React, { Component } from 'react'
import { withStyles } from '@material-ui/core/styles'
import { connect } from 'react-redux'
import _ from 'lodash'
import IconButton from '@material-ui/core/IconButton'
import Avatar from '@material-ui/core/Avatar'
import Button from '@material-ui/core/Button'
import Menu from '@material-ui/core/Menu'
import Card from '@material-ui/core/Card'
import CardActions from '@material-ui/core/CardActions'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'

import styles from './UserMenuStyles'

class UserMenu extends Component {
  handleUserLogout () {
    const { handleUserMenuClose, handleMainMenuClose } = this.props
    handleMainMenuClose()
    handleUserMenuClose('/logout')
  }

  renderUserMenuIconButton (userMenuId) {
    const {
      classes,
      authentication: { username },
      userMenuAnchorEl,
      handleUserMenuOpen
    } = this.props
    return (
      <IconButton aria-owns={userMenuAnchorEl ? userMenuId : null} aria-haspopup='true' color='inherit'
        onClick={handleUserMenuOpen}>
        <Avatar className={classes.avatar}>{_.toUpper(username[0])}</Avatar>
      </IconButton>
    )
  }

  renderUserMenuItems () {
    return (
      <Button fullWidth color='primary' variant='raised' onClick={this.handleUserLogout.bind(this)}>Sair</Button>
    )
  }

  renderUserMenu (userMenuId) {
    const {
      classes,
      authentication: { username },
      userMenuAnchorEl,
      handleUserMenuClose
    } = this.props
    return (
      <Menu
        id={userMenuId}
        anchorEl={userMenuAnchorEl}
        open={Boolean(userMenuAnchorEl)}
        onClose={handleUserMenuClose}
        className={classes.userMenu}>
        <Card elevation={0}>
          <CardContent>
            <Typography variant='body2' color='textSecondary'>{username}</Typography>
          </CardContent>
          <CardActions className={classes.row}>
            {this.renderUserMenuItems()}
          </CardActions>
        </Card>
      </Menu>
    )
  }

  render () {
    const userMenuId = 'user-menu'
    const { classes, authentication } = this.props
    return authentication && authentication.isUserAuthenticated && (
      <div className={classes.row}>
        {this.renderUserMenuIconButton(userMenuId)}
        {this.renderUserMenu(userMenuId)}
      </div>
    )
  }
}

const materialUIEnhanced = withStyles(styles)(UserMenu)

const mapStateToProps = ({ authentication }) => ({ authentication })

export default connect(mapStateToProps)(materialUIEnhanced)
