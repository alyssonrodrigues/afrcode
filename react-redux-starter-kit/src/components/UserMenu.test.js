import React from 'react'
import { shallow } from 'enzyme'

import { UserMenu } from './UserMenu'

describe('UserMenu', () => {
  it('UserMenu render deveria ocorrer sem falhas!', () => {
    const componentWrapper = shallow(
      <UserMenu classes={{}} theme={{}} userMenuAnchorEl={'id'}
        authentication={{isUserAuthenticated: true, username: 'josejoao'}}
        handleUserMenuOpen={jest.fn()} handleUserMenuClose={jest.fn()} />
    )
    expect(componentWrapper).toMatchSnapshot()
  })
})
