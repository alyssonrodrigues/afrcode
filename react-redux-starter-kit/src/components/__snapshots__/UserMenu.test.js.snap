// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMenu UserMenu render deveria ocorrer sem falhas! 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserMenu
    authentication={
        Object {
            "isUserAuthenticated": true,
            "username": "josejoao",
          }
    }
    classes={Object {}}
    handleUserMenuClose={[Function]}
    handleUserMenuOpen={[Function]}
    theme={Object {}}
    userMenuAnchorEl="id"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(IconButton)
          aria-haspopup="true"
          aria-owns="user-menu"
          color="inherit"
          onClick={[Function]}
>
          <WithStyles(Avatar)
                    className={undefined}
          >
                    J
          </WithStyles(Avatar)>
</WithStyles(IconButton)>,
        <WithStyles(Menu)
          anchorEl="id"
          className={undefined}
          id="user-menu"
          onClose={[Function]}
          open={true}
>
          <WithStyles(Card)
                    elevation={0}
          >
                    <WithStyles(CardContent)>
                              <WithStyles(Typography)
                                        color="textSecondary"
                                        variant="subtitle2"
                              >
                                        josejoao
                              </WithStyles(Typography)>
                    </WithStyles(CardContent)>
                    <WithStyles(CardActions)
                              className={undefined}
                    >
                              <WithStyles(Button)
                                        color="primary"
                                        fullWidth={true}
                                        onClick={[Function]}
                                        variant="contained"
                              >
                                        Sair
                              </WithStyles(Button)>
                    </WithStyles(CardActions)>
          </WithStyles(Card)>
</WithStyles(Menu)>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-haspopup": "true",
          "aria-owns": "user-menu",
          "children": <WithStyles(Avatar)
            className={undefined}
>
            J
</WithStyles(Avatar)>,
          "color": "inherit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "J",
            "className": undefined,
          },
          "ref": null,
          "rendered": "J",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "anchorEl": "id",
          "children": <WithStyles(Card)
            elevation={0}
>
            <WithStyles(CardContent)>
                        <WithStyles(Typography)
                                    color="textSecondary"
                                    variant="subtitle2"
                        >
                                    josejoao
                        </WithStyles(Typography)>
            </WithStyles(CardContent)>
            <WithStyles(CardActions)
                        className={undefined}
            >
                        <WithStyles(Button)
                                    color="primary"
                                    fullWidth={true}
                                    onClick={[Function]}
                                    variant="contained"
                        >
                                    Sair
                        </WithStyles(Button)>
            </WithStyles(CardActions)>
</WithStyles(Card)>,
          "className": undefined,
          "id": "user-menu",
          "onClose": [Function],
          "open": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(CardContent)>
                <WithStyles(Typography)
                                color="textSecondary"
                                variant="subtitle2"
                >
                                josejoao
                </WithStyles(Typography)>
</WithStyles(CardContent)>,
              <WithStyles(CardActions)
                className={undefined}
>
                <WithStyles(Button)
                                color="primary"
                                fullWidth={true}
                                onClick={[Function]}
                                variant="contained"
                >
                                Sair
                </WithStyles(Button)>
</WithStyles(CardActions)>,
            ],
            "elevation": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Typography)
                  color="textSecondary"
                  variant="subtitle2"
>
                  josejoao
</WithStyles(Typography)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "josejoao",
                  "color": "textSecondary",
                  "variant": "subtitle2",
                },
                "ref": null,
                "rendered": "josejoao",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Button)
                  color="primary"
                  fullWidth={true}
                  onClick={[Function]}
                  variant="contained"
>
                  Sair
</WithStyles(Button)>,
                "className": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Sair",
                  "color": "primary",
                  "fullWidth": true,
                  "onClick": [Function],
                  "variant": "contained",
                },
                "ref": null,
                "rendered": "Sair",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(IconButton)
            aria-haspopup="true"
            aria-owns="user-menu"
            color="inherit"
            onClick={[Function]}
>
            <WithStyles(Avatar)
                        className={undefined}
            >
                        J
            </WithStyles(Avatar)>
</WithStyles(IconButton)>,
          <WithStyles(Menu)
            anchorEl="id"
            className={undefined}
            id="user-menu"
            onClose={[Function]}
            open={true}
>
            <WithStyles(Card)
                        elevation={0}
            >
                        <WithStyles(CardContent)>
                                    <WithStyles(Typography)
                                                color="textSecondary"
                                                variant="subtitle2"
                                    >
                                                josejoao
                                    </WithStyles(Typography)>
                        </WithStyles(CardContent)>
                        <WithStyles(CardActions)
                                    className={undefined}
                        >
                                    <WithStyles(Button)
                                                color="primary"
                                                fullWidth={true}
                                                onClick={[Function]}
                                                variant="contained"
                                    >
                                                Sair
                                    </WithStyles(Button)>
                        </WithStyles(CardActions)>
            </WithStyles(Card)>
</WithStyles(Menu)>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "aria-haspopup": "true",
            "aria-owns": "user-menu",
            "children": <WithStyles(Avatar)
              className={undefined}
>
              J
</WithStyles(Avatar)>,
            "color": "inherit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "J",
              "className": undefined,
            },
            "ref": null,
            "rendered": "J",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "anchorEl": "id",
            "children": <WithStyles(Card)
              elevation={0}
>
              <WithStyles(CardContent)>
                            <WithStyles(Typography)
                                          color="textSecondary"
                                          variant="subtitle2"
                            >
                                          josejoao
                            </WithStyles(Typography)>
              </WithStyles(CardContent)>
              <WithStyles(CardActions)
                            className={undefined}
              >
                            <WithStyles(Button)
                                          color="primary"
                                          fullWidth={true}
                                          onClick={[Function]}
                                          variant="contained"
                            >
                                          Sair
                            </WithStyles(Button)>
              </WithStyles(CardActions)>
</WithStyles(Card)>,
            "className": undefined,
            "id": "user-menu",
            "onClose": [Function],
            "open": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(CardContent)>
                  <WithStyles(Typography)
                                    color="textSecondary"
                                    variant="subtitle2"
                  >
                                    josejoao
                  </WithStyles(Typography)>
</WithStyles(CardContent)>,
                <WithStyles(CardActions)
                  className={undefined}
>
                  <WithStyles(Button)
                                    color="primary"
                                    fullWidth={true}
                                    onClick={[Function]}
                                    variant="contained"
                  >
                                    Sair
                  </WithStyles(Button)>
</WithStyles(CardActions)>,
              ],
              "elevation": 0,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Typography)
                    color="textSecondary"
                    variant="subtitle2"
>
                    josejoao
</WithStyles(Typography)>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "josejoao",
                    "color": "textSecondary",
                    "variant": "subtitle2",
                  },
                  "ref": null,
                  "rendered": "josejoao",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Button)
                    color="primary"
                    fullWidth={true}
                    onClick={[Function]}
                    variant="contained"
>
                    Sair
</WithStyles(Button)>,
                  "className": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sair",
                    "color": "primary",
                    "fullWidth": true,
                    "onClick": [Function],
                    "variant": "contained",
                  },
                  "ref": null,
                  "rendered": "Sair",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
